### ISSUE TITLE
Killing rectangular region

### ISSUE THREAD

**Issue by BloodOrange** [syl20bnr/spacemacs#3278]
I have an issue about the new rectangular region of emacs 24.4. I use emacs 24.5.

With C-x <SPC> you use a rectangular marker. Normally, with C-w you cut the rectangular region but Spacemacs cuts the region like it was a normal marker.

My english is very poor, so this is an example :
|toto one
 toto two
 toto three

| is my cursor. I press C-x <SPC>, then I move my cursor
toto  one
toto  two
toto| three

So, when I pressed C-w I have
three

But I would like
one
two
three

PS: It's my first message, so I want thank you for your very good job. I like Spacemacs :)

**Comment by bmag** [comment]
1) Can you provide the output of `SPC h d s` (or `M-m h d s` if you use holy-mode)? It will give us more info about your setup.

2) You're describing a very weird behavior. Selecting a rectangular region should be `C-x SPC`, not `C-x` alone. What `C-w` depends on the state you are in (emacs state, insert state, normal state, etc.) and your editing style (vim, emacs, hybrid). Rectangular selection and deletion works just fine for me.

3) Can you tell as in what state you are (`SPC h d v evil-state` or `C-h v evil-state`) and what is the major mode (`SPC h d v major-mode` or `C-h v major-mode`) of the buffer when you encounter this behavior?

**Comment by BloodOrange** [comment]
1) #### System Info
- OS: cygwin
- Emacs: 24.5.1
- Spacemacs: 0.104.2
- Spacemacs branch: master (rev. 074f425)
- Distribution: spacemacs
- Layers:
```elisp
(auto-completion better-defaults emacs-lisp git org version-control gtags)
```

2) I wrote SPC with bracket but it miss in the post. Sorry, 
With C-x SPC I have the rectangle region but when I pressed C-w, emacs cut the text like it was a normal region.

3) I use Spacemacs with holy-mode. The active state is "emacs".

I try on two computers. The first with cygwin on windows and the second with kubuntu. I have the same result on both computers.

**Comment by bmag** [comment]
I was able to reproduce the bug. Turns out the bug is caused by the layer `better-defaults`. The layer binds `C-w` to `spacemacs/backward-kill-word-or-region`, which doesn't handle a rectangular selection correctly. The original binding for `C-w` is `kill-region`.
As a temporary work-around, you can bind `C-w` to `kill-region` in your `dotspacemacs/user-config`:
```elisp
(global-set-key (kbd "C-w") #'kill-region)
```
I'll try later to fix the bug in `spacemacs/backward-kill-word-or-region`, unless you want to try fixing it yourself. It shouldn't be too hard.

**Comment by BloodOrange** [comment]
I fixed it but I don't know where I can put the code. 

So, this is :
(defun spacemacs/backward-kill-word-or-region (&optional arg)
  "Calls `kill-region' when a region is active and
`backward-kill-word' otherwise. ARG is passed to
`backward-kill-word' if no region is active."
  (interactive "p")
  (if (region-active-p)
      (if rectangle-mark-mode
          (kill-rectangle (region-beginning) (region-end))
        (kill-region (region-beginning) (region-end))
        )
    (backward-kill-word arg)))

Maybe you can write it more efficient but I am not conformtable with elisp. I did not find a lot of documentation about rectangle-mark and I don't understand why when you bind the "kill-region" that works but when you call this function with elisp that doen't. I think it's beceause there are two functions. The first interactive was called when you bind a key and the second was called by elisp code.

I hope this can help you. :)

**Comment by bmag** [comment]
@BloodOrange there is a difference between calling a command interactively (e.g. via `M-x` or a key-binding) and calling it from elisp code. The function `call-interactively` calls a command as if the user invoked it interactively. For more technical info, see sections [20.2](http://www.gnu.org/software/emacs/manual/html_node/elisp/Defining-Commands.html#Defining-Commands) and [20.3](http://www.gnu.org/software/emacs/manual/html_node/elisp/Interactive-Call.html) of the elisp manual.
Anyway, I fixed the bug in #3316.

**Comment by BloodOrange** [comment]
Ok,

Thank you for your explaination.
